pkg/executor/executor.go

```diff
@@ -410,19 +411,65 @@ type DDLJobRetriever struct {
 	activeRoles    []*auth.RoleIdentity
 	cacheJobs      []*model.Job
 	TZLoc          *time.Location
+	extractor      base.MemTablePredicateExtractor
 }
 
 func (e *DDLJobRetriever) initial(txn kv.Transaction, sess sessionctx.Context) error {
-	m := meta.NewMeta(txn)
-	jobs, err := ddl.GetAllDDLJobs(sess)
-	if err != nil {
-		return err
+	skipRunningJobs := false
+	skipHistoryJobs := false
+	schemaFilter := "1"
+	schemaIDSet := set.NewInt64Set()
+	tableNames := set.NewStringSet()
+
+	ex, ok := e.extractor.(*plannercore.InfoSchemaDDLExtractor)
+	if ok {
+		if states, ok := ex.ColPredicates["state"]; ok {
+			skipHistoryJobs = true
+			skipRunningJobs = true
+			states.IterateWith(func(s string) {
+				ss := strings.ToLower(s)
+				if ss == "cancelled" || ss == "synced" {
+					skipHistoryJobs = false
+				} else {
+					skipRunningJobs = false
+				}
+			})
+		}
+
+		// build schema_ids filter
+		if sIDs, hasPredicates := ex.ListSchemaIDs(e.is); len(sIDs) > 0 {
+			var builder strings.Builder
+			for _, dbid := range sIDs {
+				schemaIDSet.Insert(dbid)
+				builder.WriteString(fmt.Sprintf("schema_ids LIKE '%%%d%%' OR ", dbid))
+			}
+			schemaFilter = builder.String()
+			schemaFilter = schemaFilter[:len(schemaFilter)-4]
+		} else if hasPredicates {
+			// This means all schemas in predicates are not found, just skip
```

Comment:
Does it mean the schemas are dropped, or not yet created? So we should not skip them

Reply:
You are right, for `create table` and `create database`, the corresponding info of schema/table cannot be found in infoschema. Maybe we can't apply any filtering on running job table.